### ISSUE TITLE
Writing & (ampersand) causes error

### ISSUE THREAD

**Issue by nicojx** [JanMarvin/openxlsx2#467]
Writing a value with an ampersand (&) causes an error when opening the workbook:

```R
wb_workbook()$
  add_worksheet()$
  add_data(dims = "A1", x = "A & B")$
  open()
```

The 'repair' error message says: 
`Replaced Part: /xl/worksheets/sheet1.xml part with XML error.  Illegal name character. Line 1, column 1063.`

**Comment by nicojx** [comment]
I should add, this happens with PR#461 -- I don't remember the issue existing previously, though maybe I just missed it.

**Comment by JanMarvin** [comment]
Hm, my fix in #435 was actually bad. Thought you had double checked it 😉

**Comment by JanMarvin** [comment]
The initial idea was alright, I simply forgot that pugixml does not reduce `&amp;` and that we have to do it our self. This is now in #468

**Comment by nicojx** [comment]
Amazing, thanks! :)

**Comment by JanMarvin** [comment]
Alright, a couple of hours later the penny dropped. The entire thing was caused by a default escape to xml when printing inline strings from the worksheet (this escaped the already escaped `&amp;` to `&amp;amp;` which pugi correctly unescaped to `&amp;` afterwards). Ofc mighty pugi did everything correct, I just messed up :see_no_evil: 

Guess I start merging open pull requests tomorrow so it might be in main on Monday (I like to stretch them a bit to reduce the confusion if something breaks).
